### ISSUE TITLE
Pull request from fork repository to main one

### ISSUE THREAD

**Issue by Cosmos491** [ubsuny/CP1-24-HW2#73]
Hi all, I need some guidance regarding what to do after completing the homework assignment 2, as in I have written the code and the documentation. Now I want to create a pull request but I am not sure if I am correctly doing so as it is not showing on the main branch. Can someone help me on this?

**Comment by iglesias-cardinale** [comment]
Did you write your code in a codespace on your forked repository?

**Comment by Cosmos491** [comment]
I wrote the code in codespace, and the documentation in my forked repository.

**Comment by iglesias-cardinale** [comment]
Then there should be an option in your repository to create a pull request. Once you click on "create pull request" you should be able to click the branch you want to pull from. 

I might recommend that you split your work into multiple branches and do multiple pull requests so that you can keep track of everything, make sure your reviewer doesn't have so much to review, and get more homework points.

**Comment by Cosmos491** [comment]
Ohh, I see. On my branch, it also asks me to update my branch, or discard commits. Should I update it and then click on the pull request?

**Comment by iglesias-cardinale** [comment]
Yes, you should sync your fork before you do your pull request. It'll just make sure your branch is up to date with the main repository.

**Comment by Cosmos491** [comment]
Thanks Vince, I really appreciate your help on this. Can you also tell me how to push and pull. I am guessing I should sync it first, then do the push request, and then the pull request?

<img width="924" alt="Screenshot 2024-10-06 at 8 27 32 PM" src="https://github.com/user-attachments/assets/fab78418-a6b7-4909-8be6-9ffef3108cbe">


-Vedant

**Comment by ojha-aditya** [comment]
Sync would do it for you, push changes to the branch on which your codespace is set up and pull the changes done on the main repository by someone to the same. 
After committing your codespace changes and syncing, it would be possible to create a pull request from here to the main repository.
